PhysicsTools/JetMCAlgos/plugins/TauGenJetProducer.cc

```diff
@@ -41,6 +40,20 @@ void TauGenJetProducer::produce(edm::StreamID, Event& iEvent, const EventSetup&
     GenParticleRefVector descendents;
     findDescendents(*iTau, descendents, 1);
 
+    if (descendents.empty()) {
```

Comment:
Uhm, shouldn't this go after the check for the status 2 taus in the descendents?
@alebihan @mbluj 

(Maybe @mbluj could cherry pick this commit and take care of making a PR with the whole fix, so that he can include also the modified status all at once)

Reply:
The taus causing the problem were never in the descendants, they always came from the main list.